src/screens/TopicPageScreen/index.js

```diff
@@ -413,31 +422,35 @@ const TopicPageScreen = (props) => {
         isFollow={isFollow}
         getTopicDetail={getTopicDetail}
       />
-      <TiktokScroll
-        ref={listRef}
-        contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
-        data={feeds}
-        onEndReach={onEndReach}
-        onRefresh={onRefresh}
-        refreshing={loading}
-        renderItem={renderItem}
-        onScroll={handleScroll}
-        scrollEventThrottle={16}
-        showSearchBar={true}
-        searchHeight={isHeaderHide ? headerHideHeight : headerShowHeight}
-        snap
-        contentOffset={{x: 0, y: topPosition}}
-        contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
-        onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
-      />
+
+      <View
+        style={{
+          marginTop: isHeaderHide ? -dimen.size.TOPIC_FEED_HEADER_HEIGHT : 0
+        }}>
+        <TiktokScroll
+          ref={listRef}
+          contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
+          data={feeds}
+          onEndReach={onEndReach}
+          onRefresh={onRefresh}
+          refreshing={loading}
+          renderItem={renderItem}
+          onScroll={handleScroll}
+          scrollEventThrottle={16}
+          showSearchBar={true}
+          snap
+          contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
+          onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
+        />
+      </View>
       <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} />
       <BlockComponent
         ref={refBlockComponent}
         refresh={onBlockCompleted}
         refreshAnonymous={onDeleteBlockedPostCompleted}
         screen="topic_screen"
       />
-    </SafeAreaProvider>
+    </View>
   );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The main change here is the wrapping of the `TiktokScroll`, `ButtonAddPostTopic`, and `BlockComponent` components inside a `View` component. This should not cause any issues as long as the styles are adjusted accordingly. However, the `searchHeight` prop has been removed from `TiktokScroll` and replaced with `marginTop` style in the wrapping `View`. Make sure this doesn't affect the layout or functionality of the `TiktokScroll`.

```diff
+ <View
+   style={{
+     marginTop: isHeaderHide ? -dimen.size.TOPIC_FEED_HEADER_HEIGHT : 0
+   }}>
+   <TiktokScroll
+     ref={listRef}
+     contentHeight={dimen.size.TOPIC_CURRENT_ITEM_HEIGHT + normalizeFontSizeByWidth(4)}
+     data={feeds}
+     onEndReach={onEndReach}
+     onRefresh={onRefresh}
+     refreshing={loading}
+     renderItem={renderItem}
+     onScroll={handleScroll}
+     scrollEventThrottle={16}
+     showSearchBar={true}
+     snap
+     contentInsetAdjustmentBehavior={feeds?.length > 1 ? 'automatic' : 'never'}
+     onMomentumScrollEnd={(event) => onWillSendViewPostTime(event, feeds)}
+   />
+ </View>
+ <ButtonAddPostTopic topicName={topicName} onRefresh={onRefresh} />
+ <BlockComponent
+   ref={refBlockComponent}
+   refresh={onBlockCompleted}
+   refreshAnonymous={onDeleteBlockedPostCompleted}
+   screen="topic_screen"
+ />
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->